components/autofill/src/db/credit_cards.rs

```diff
@@ -138,6 +135,44 @@ pub fn update_credit_card(
     Ok(())
 }
 
+/// Updates all fields including metadata - although the change counter gets
+/// slighly special treatment. Suitable for internal use (eg, by Sync)
+pub(crate) fn update_internal_credit_card(
+    tx: &Transaction<'_>,
+    card: &InternalCreditCard,
+    flag_as_changed: bool,
+) -> Result<()> {
+    let change_counter_increment = flag_as_changed as u32; // will be 1 or 0
```

Comment:
I might be missing something here but under what conditions would we make an update call while not wanting to increment the `sync_change_counter`?

Reply:
Sync calls this function when updating due to an incoming record - the change counter should not increment in that case